### PR TITLE
Change default read timeout to 1 second

### PR DESCRIPTION
**Pull request by dain**
I'm not sure if the values are correct, but this shows how the default config system works

### CODE REVIEW

**Comment by dain** [comment]
Updated to new configuration defaults system

File: presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java [hunk 1]
```diff
@@ -162,7 +164,11 @@ protected void setup(Binder binder)
 
         // exchange client
         binder.bind(new TypeLiteral<Supplier<ExchangeClient>>() {}).to(ExchangeClientFactory.class).in(Scopes.SINGLETON);
-        httpClientBinder(binder).bindHttpClient("exchange", ForExchange.class).withTracing();
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java]: did you remove `withTracing()` on purpose?

File: presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java [hunk 2]
```diff
@@ -162,7 +164,11 @@ protected void setup(Binder binder)
...
+        httpClientBinder(binder).bindHttpClient("exchange", ForExchange.class)
+                .withConfigDefaults(c -> {
+                    c.setReadTimeout(new Duration(1, SECONDS));
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java]: What's the current timeout? 1sec seems pretty aggressive

**Comment by cberner** [comment]
Looks good, but I had some questions

File: presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java (hunk 1 above)
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java]: No. Fixed it.

**Comment by dain** [comment]
Ready to go, but waiting on a review of https://github.com/airlift/airlift/pull/273 by @electrum

File: presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java [hunk 3]
```diff
@@ -170,7 +172,12 @@ protected void setup(Binder binder)
 
         // exchange client
         binder.bind(new TypeLiteral<Supplier<ExchangeClient>>() {}).to(ExchangeClientFactory.class).in(Scopes.SINGLETON);
-        httpClientBinder(binder).bindHttpClient("exchange", ForExchange.class).withTracing();
+        httpClientBinder(binder).bindHttpClient("exchange", ForExchange.class)
+                .withConfigDefaults(c -> {
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java]: `config` instead of `c` for your variable name

**Comment by dain** [comment]
@cberner I added a new commit "Update to new config binder API" and changed the defaults for http client a bit

**Comment by cberner** [comment]
looks good

**Pull request merged**